### ISSUE TITLE
eager_load tries to load tag_names on Rails 4.2

### ISSUE THREAD

**Issue by jellyfunk** [pat/gutentag#67]
Under Rails 4.2, queries that eager-load a tagged model try to load a non-existent  `tag_names` column from the underlying table. 

e.g for the usual trivial blog models:
```ruby
class Post < ActiveRecord::Base
  has_many :comments
  Gutentag::ActiveRecord.call(self)
end

class Comment < ActiveRecord::Base
  belongs_to :post
end
```

calling `Comment.eager_load(:post)`  or `Comment.references(:post).includes(:post)`generates the following SQL:
```sql
SELECT "comments"."id" AS t0_r0,
       "comments"."post_id" AS t0_r1,
       "comments"."body" AS t0_r2,
       "comments"."created_at" AS t0_r3,
       "comments"."updated_at" AS t0_r4,
       "posts"."id" AS t1_r0,
       "posts"."title" AS t1_r1,
       "posts"."body" AS t1_r2,
       "posts"."created_at" AS t1_r3,
       "posts"."updated_at" AS t1_r4,
       "posts"."tag_names" AS t1_r5
FROM "comments" LEFT OUTER JOIN "posts" ON "posts"."id" = "comments"."post_id"
```

 The same queries work under Rails 5.2, so it seems to be a difference in the behaviour of the Attributes API in 4.2

Changing `lib/gutentag/active_record.rb` to treat 4.2 the same way as <=4.1 seemed to resolve it in casual testing via the Rails console.

**Comment by pat** [comment]
I've just pushed a fix for this - if you've got the chance to test it and confirm everything's working well for you, that'd be great - and then I can push out a new release with a touch more confidence 😄. I have added a regression test for the issue you hit, and the test suite does run across multiple Rails versions, so I'm pretty fine with where things stand, but the confirmation wouldn't hurt.

```ruby
gem "gutentag",
  :git    => "https://github.com/pat/gutentag.git",
  :branch => "master",
  :ref    => "303d70721c"
```

**Comment by jellyfunk** [comment]
My tests are all green again 👍 

Thanks for looking at this so quickly!

**Comment by pat** [comment]
Greatly appreciate you checking - I've just published 2.5.2 with the fix :)
